File: backend-java/src/main/java/com/citypilot/parking/repositories/OwnerRepository.java
package com.citypilot.parking.repositories;

import java.util.Map;

import static com.citypilot.parking.utils.JsonUtil.castMap;
import static com.citypilot.parking.utils.JsonUtil.object;

public class OwnerRepository {
    public synchronized Map<String, Object> getOwnerProfile() {
        return DataStore.OWNER;
    }

    public synchronized Map<String, Object> updateRentStatus(boolean renting) {
        DataStore.OWNER.put("renting", renting);
        castMap(DataStore.OWNER.get("spot")).put("status", renting ? "出租中" : "已暂停");
        return DataStore.OWNER;
    }

    public synchronized Map<String, Object> createOwnerSpot(Map<String, Object> input) {
        Map<String, Object> spot = object(
                "name", stringOrDefault(input.get("name"), "新增共享车位"),
                "window", stringOrDefault(input.get("window"), "工作日 08:30 - 18:00"),
                "price", stringOrDefault(input.get("price"), "¥4 / 时"),
                "status", "出租中"
        );
        DataStore.OWNER.put("spot", spot);
        DataStore.OWNER.put("renting", true);
        return spot;
    }

    private String stringOrDefault(Object value, String fallback) {
        if (value == null || String.valueOf(value).trim().isEmpty()) return fallback;
        return String.valueOf(value);
    }
}

File: backend/repositories/owner-repository.js
const store = require("../data");

function getOwnerProfile() {
  return store.owner;
}

function updateRentStatus(renting) {
  store.owner.renting = Boolean(renting);
  store.owner.spot.status = store.owner.renting ? "出租中" : "已暂停";
  return store.owner;
}

function createOwnerSpot(input) {
  store.owner.spot = {
    name: input.name || "新增共享车位",
    window: input.window || "工作日 08:30 - 18:00",
    price: input.price || "¥4 / 时",
    status: "审核通过"
  };
  store.owner.renting = true;
  return store.owner.spot;
}

module.exports = {
  getOwnerProfile,
  updateRentStatus,
  createOwnerSpot
};
